### ISSUE TITLE
calc should use wavelength in angstrom

### ISSUE THREAD

**Issue by prjemian** [bluesky/hklpy#17]
Following the [E6C example](https://repo.or.cz/hkl.git/blob/8bc6ca7f36ea7cb6a18b3758fabaca29dac259fc:/tests/bindings/crystal.ini) from the *hkl* source code, wavelength should be specified in angstrom.  The problem is in this *hklpy* package, conversion from energy to wavelength uses nm.  https://github.com/bluesky/hklpy/blob/169dd09470c6d6f67e7be076970451301d323662/hkl/calc.py#L143-L150

The voltage-wavelength product is specified as `NM_KEV` which expresses a clear intent of this package to use nm.  https://github.com/bluesky/hklpy/blob/169dd09470c6d6f67e7be076970451301d323662/hkl/calc.py#L16

However, when energy is specified, it results in the wavelength expressed in nm, and an incorrect orientation matrix is computed.  The underlying *hkl* code uses angstrom.

Should discuss before making a change here.

**Comment by prjemian** [comment]
@ambarb : Would you be affected if I change this package to use wavelength in angstrom rather than nm?  Read the issue above for justification.  This is disturbing that no one has commented on this previously.

**Comment by prjemian** [comment]
Who else might be affected by this change?

**Comment by ambarb** [comment]
@prjemian I think changing to angstroms are great. We asked a lot about this in the beginning.  Instead, CSX decided to use angstroms for the lattice constants, but we take care of the conversion issue by setting the hklpy device energy to something that is not keV.  So if we every change the energy for the calculations (either because we want to calculate something or we have changed the incident energy), we have to do things manually, like for spec  (e.g., LAMBDA = 1.54). 

Other things we need to consider are:
* ability add/subtract energy offset instead of making people adjust lattice constants
* ability to convert the RBV of beamline energy appropriately (soft x-ray beamlines at NSLS-II use eV, not keV). In other words, rely on EPICS UNITS to decide how to convert.

If we continue to require a manual setting of the hklpy calculation energy, then these things are transparent.  I am just anticipating something more sophisticated as I am not sure how much you want to change.

I cannot find GitHub handles for everyone that may have some preference, but I will forward this thread.

**Comment by prjemian** [comment]
Those are good considerations for future modifications.  Units conversion can be accomplished by use of the[cfunits](https://bitbucket.org/cfpython/cfunits-python/src/master/) package which uses [Unidata UDUNITS](https://www.unidata.ucar.edu/software/udunits/).  This requires the EPICS record to have a `/EGU` field value compatible with UDUNITS.  Conversion of `eV` to `keV` is compatible.

**Comment by prjemian** [comment]
This issue should focus on just the correction of the wavelength units.  Here is an example of why.  In one of the supplied examples, for the tardis diffractometer, there is a jupyter notebook that demonstrates various operations with the support code.  When it comes to wavelength and energy, this part is easily seen as clearly wrong:  https://github.com/bluesky/hklpy/blob/066d20bf4c34e8312c007cd20a3dea987ec81629/examples/tardis_example.ipynb

![Clipboard01](https://user-images.githubusercontent.com/2279984/75358725-fbad9b80-5878-11ea-8204-03a0c73e09d8.jpg)

The energy is wrong by a factor 10, the exact conversion constant for nm to angstrom.  This error should have been caught when this example was created.  For a wavelength of 1.6 angstrom, the energy is in the range of 7 keV, not 0.7 keV.  The conversion constant and reported usage in the hklpy code are wrong since the *hkl* package, and the lattice parameters in this package use angstrom units.  Additional corrections will need to be made in all examples that describe wavelength and energy.

**Comment by ambarb** [comment]
@prjemian this was just a development notebook from the very beginning to test the implementation and not what we do now.

`tardis.calc.energy = (pgm.energy.setpoint.value + 3.1)/10000` is what is used.  We are able to drive to diffraction peak.

**Comment by prjemian** [comment]
The units of `pgm.energy` are *eV*, correct?  Your conversion constant of *10^4* accounts for the additional error cited by this issue.

What are the values of `tardis.calc.energy` and  `tardis.calc.wavelength` after you set `tardis.calc.energy`?

**Comment by prjemian** [comment]
I'll make up a branch with changes and submit a PR for your review.  Add Stuart WIlkins or cmazzoli as additional possible reviewers.

**Comment by ambarb** [comment]
yes, `pgm.energy` is has units of eV for CSX.  In the above example, our energy calibration was incorrect by 3.1 eV.  

`tardis.calc.energy` = 0.09311
`tardis.calc.wavelength` = 13.31588

**Comment by prjemian** [comment]
Confirming that `pgm.energy.setpoint.value` is actually 90.01 eV, correct?  Or is it 930 eV?

Are you certain `tardis.calc.energy` has keV units?

I think your factor of *10^4* is off by 10 but results in `tardis.calc.wavelength` being computed to the correct value.  Agree?

**Comment by prjemian** [comment]
I fixed my math above to get 928 eV.

**Comment by prjemian** [comment]
OK, we got sidetracked and dropped this issue on the floor.  The wavelength issue must be corrected (bit by it at APS two weeks ago) but changes in the examples should be addressed in a different issue/branch/PR.

I'll start a new branch from today's master for just the wavelength change.  I'll leave the existing branch ([`17-wavelength-units`](https://github.com/bluesky/hklpy/branches)) open for code references.  Once we have decided how to handle the changes to the examples, the `17-wavelength-units` branch can be closed without merge.
